source/UF/Truncations.lagda

```diff
@@ -0,0 +1,428 @@
+   where
+    I : (x' : X) (p : x ＝ x')
+      → (∣ x ∣[ succ n ] , refl-trunc-id-family)
+       ＝[ Σ (trunc-id-family-type) ]
+        (∣ x' ∣[ succ n ] , trunc-id-backward-map x' ∣ p ∣[ n ])
+    I x' refl = refl
+    II : (x' : X) (q' : ∥ x ＝ x' ∥[ n ])
+       → (∣ x ∣[ succ n ] , refl-trunc-id-family)
+        ＝[ Σ (trunc-id-family-type) ]
+         (∣ x' ∣[ succ n ] , trunc-id-backward-map x' q')
+    II x' = ∥∥ₙ-ind (λ s → hlevel-closed-under-Σ (succ n)
+                            trunc-id-family-type (∥∥ₙ-h-level (succ n))
+                            (λ v → hlevels-are-upper-closed n
+                                    (trunc-id-family-type v)
+                                    (trunc-id-family-level v))
+                            (∣ x ∣[ succ n ] , refl-trunc-id-family)
+                            (∣ x' ∣[ succ n ] , trunc-id-backward-map x' s))
+                     (I x')
+    III : (x' : X) (q : trunc-id-family-type ∣ x' ∣[ succ n ])
+        → (∣ x ∣[ succ n ] , refl-trunc-id-family)
+          ＝[ Σ (trunc-id-family-type) ]
+          (∣ x' ∣[ succ n ] , q)
+    III x' q = transport (λ - → (∣ x ∣[ succ n ] , refl-trunc-id-family)
+                                ＝[ Σ (trunc-id-family-type) ]
+                                (∣ x' ∣[ succ n ] , -))
+                         (trunc-id-back-is-retraction x' q)
+                         (II x' (trunc-id-forward-map x' q))
+    IV : (v : ∥ X ∥[ succ n ]) (q : trunc-id-family-type v)
+       → (∣ x ∣[ succ n ] , refl-trunc-id-family) ＝ (v , q)
+    IV =
+     ∥∥ₙ-ind (λ s → hlevel-closed-under-Π (succ n) 
+                     (λ q → (∣ x ∣[ succ n ] , refl-trunc-id-family) ＝ (s , q))
+                     (λ q → hlevel-closed-under-Σ (succ (succ n)) 
+                             trunc-id-family-type (hlevels-are-upper-closed
+                                                   (succ n) ∥ X ∥[ succ n ]
+                                                   (∥∥ₙ-h-level (succ n)))
+                                                  (λ v → hlevels-are-upper-closed
+                                                    (succ n)
+                                                    (trunc-id-family-type v)
+                                                    (hlevels-are-upper-closed n
+                                                     (trunc-id-family-type v)
+                                                     (trunc-id-family-level v)))
+                                                  (∣ x ∣[ succ n ]
+                                                   , refl-trunc-id-family)
+                                                  (s , q)))
+             III
+    center-Q : is-central (Σ (trunc-id-family-type))
+                          (∣ x ∣[ succ n ] , refl-trunc-id-family)
+    center-Q (v , q) = IV v q 
+
+ trunc-identity-characterization : {𝓤 : Universe} {X : 𝓤 ̇} {n : ℕ}
+                                 → (ua : Univalence)
```

Comment:
We only need univalence for 𝓤 right? If so, I think having `is-univalent 𝓤` is a nicer assumption to make everywhere.